### PR TITLE
Adds LX and REF for transactions

### PR DESCRIPTION
**Pull request by rposborne**
This resolves all stdout warnings of
Identifier: LX not handled in transaction loop.
Identifier: REF not handled in transaction loop.

While these fields don't seem super useful no warnings is quite nice.

This also adds a nevada medicaid sample found here https://www.medicaid.nv.gov/Downloads/provider/Sample_835_File.pdf

Specs are here
https://www.cms.gov/medicare/billing/electronicbillingeditrans/downloads/835-flatfile.pdf

Grep for EV, and TJ
[AT-5936]

[AT-5936]: https://affect-thera.atlassian.net/browse/AT-5936?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: edi_835_parser/loops/transaction.py [hunk 1]
```diff
@@ -123,6 +126,16 @@ def other_payee_identification(self) -> Optional[ReferenceSegment]:
...
 
+	@property
+	def reference_identification_number(self) -> Optional[ReferenceSegment]:
+		for ref in self.references:
+			if ref.qualifier == 'Reference Identification Number':
+				return ref
+
+	@property
+	def production_date(self) -> datetime.datetime:
+		return self.date.date
```
Comment (rposborne) [inline edi_835_parser/loops/transaction.py:137]: Just a convenience method.

File: edi_835_parser/loops/transaction.py [hunk 2]
```diff
@@ -170,18 +183,28 @@ def build(
...
+					transaction.date = date
+					segment = None
 
 				elif identifier in cls.terminating_identifiers:
 					return transaction, segments, segment
 
 				else:
-					segment = None
-					message = f'Identifier: {identifier} not handled in transaction loop.'
+					message = f'Identifier: {identifier} not handled in transaction loop. {segment}'
```
Comment (rposborne) [inline edi_835_parser/loops/transaction.py:205]: I found seeing the segment right away **VERY** useful to go check specs.

File: edi_835_parser/segments/header_number.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+	def __init__(self, segment: str):
+		self.index = segment.split(':', 1)[0]
+		segment = segment.split(':', 1)[1]
+
+		self.segment = segment
+		segment = split_segment(segment)
+
+		self.identifier = segment[0]
+		self.number = segment[1]
```
Comment (rposborne) [inline edi_835_parser/segments/header_number.py:20]: Still feel the need for a base class here.

File: pyproject.toml [hunk 4]
```diff
@@ -12,6 +12,8 @@ python = "^3.9"
 pandas = "^2.0.3"
 pytest = "^8.1.1"
 
+[tool.poetry.dev-dependencies]
+ruff = "*"
```
Comment (rposborne) [inline pyproject.toml:16]: Snuck this in.

File: tests/conftest.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 import pytest
 
 sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+sys.path.append(os.path.join(os.path.dirname(__file__), 'utils'))
```
Comment (rposborne) [inline tests/conftest.py:7]: Open to different approaches here, but I wanted to separate the test helpers from the tests.

File: tests/utils/__init__.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	return total_interest
+
+
+def get_claim_by_control_number(transaction_set, claim_id):
+	for transaction in transaction_set.transactions:
+		for claim in transaction.claims:
+			if claim.claim.patient_control_number == claim_id:
+				return claim
+
+	assert False, f'Claim with ID {claim_id} not found'
```
Comment (rposborne) [inline tests/utils/__init__.py:44]: Added this to be easier to hard check specific values in the test suite.

File: edi_835_parser/loops/transaction.py [hunk 7]
```diff
@@ -123,6 +126,16 @@ def other_payee_identification(self) -> Optional[ReferenceSegment]:
...
 				return ref
 
+	@property
+	def reference_identification_number(self) -> Optional[ReferenceSegment]:
+		for ref in self.references:
+			if ref.qualifier == 'Reference Identification Number':
+				return ref
+
+	@property
+	def production_date(self) -> datetime.datetime:
```
Comment (curtisim0) [inline edi_835_parser/loops/transaction.py:136]:
```suggestion
	def production_date(self) -> datetime.date:
```

Comment (curtisim0) [reply edi_835_parser/loops/transaction.py:136]: right?

File: edi_835_parser/loops/transaction.py [hunk 8]
```diff
@@ -123,6 +126,16 @@ def other_payee_identification(self) -> Optional[ReferenceSegment]:
 			if ref.qualifier == 'payee identification':
 				return ref
 
+	@property
+	def reference_identification_number(self) -> Optional[ReferenceSegment]:
+		for ref in self.references:
+			if ref.qualifier == 'Reference Identification Number':
+				return ref
```
Comment (curtisim0) [inline edi_835_parser/loops/transaction.py:133]: I trust that this is the way. If we have multiple refs, are we wanting to return the moment it hits that value? or should there be a more "hardened" rule on the return. Just checking not ⚔️

File: edi_835_parser/elements/reference_qualifier.py [hunk 9]
```diff
@@ -20,16 +20,11 @@
 	'G2': 'Provider Commerical Number',
 	'HPI': 'Centers for Medicare and Medicaid Services National Provider Identifier',
 	'SY': 'Social Security Number',
+	# https://www.cms.gov/medicare/billing/electronicbillingeditrans/downloads/835-flatfile.pdf
+	'EV': 'Reference Identification Number',
```
Comment (curtisim0) [inline edi_835_parser/elements/reference_qualifier.py:24]: o my gracious linking these out would be lovely

File: .vscode/settings.json [hunk 10]
```diff
@@ -1,7 +1,6 @@
 {
-  "python.testing.pytestArgs": [
-    "tests"
-  ],
+  "python.testing.pytestArgs": ["tests"],
   "python.testing.unittestEnabled": false,
-  "python.testing.pytestEnabled": true
+  "python.testing.pytestEnabled": true,
+  "python.analysis.extraPaths": ["./tests"]
```
Comment (curtisim0) [inline .vscode/settings.json:5]: Why we committing vscode settings?

File: edi_835_parser/loops/transaction.py (hunk 2 above)
Comment (curtisim0) [reply edi_835_parser/loops/transaction.py:205]: Freaking neeeeeeeeed a better pattern here. But yes, this is gud 👍

File: edi_835_parser/segments/header_number.py (hunk 3 above)
Comment (curtisim0) [reply edi_835_parser/segments/header_number.py:20]: fo sho

File: tests/conftest.py (hunk 5 above)
Comment (curtisim0) [inline tests/conftest.py:7]: We are not pytest.ini for backwards compatibility and what not, is that correct?

Comment (curtisim0) [reply tests/conftest.py:7]: ironically these are all pytest fixtures 🤔

File: tests/utils/__init__.py (hunk 6 above)
Comment (curtisim0) [reply tests/utils/__init__.py:44]: This is great 👍

**Review by curtisim0 — APPROVED** [review]
Some thoughts, otherwise, totally fine to keep the 🚂-🚋-🚋-🚋 moving!

File: edi_835_parser/loops/transaction.py (hunk 7 above)
Comment (rposborne) [reply edi_835_parser/loops/transaction.py:136]: Right now it's parsed to a datetime.

**Pull request merged**